examples/niscope_acquire_waveform/measurement.py

```diff
@@ -132,98 +134,77 @@ def cancel_callback():
             )
         )
 
-        if len(reservation.session_info) != 1:
-            measurement_service.context.abort(
-                grpc.StatusCode.INVALID_ARGUMENT,
-                f"Unsupported number of sessions: {len(reservation.session_info)}",
+        sessions = [
+            stack.enter_context(
+                _create_niscope_session(
+                    session_info, get_grpc_device_channel(measurement_service, niscope)
+                )
+            )
+            for session_info in reservation.session_info
+        ]
+
+        for session_index, channel_list in reservation.get_sessions_and_channels_for_pins(
+            pins=pin_names
+        ):
+            session_info = reservation.session_info[session_index]
+            scope_session = sessions[session_index]
+            channel_names = session_info.channel_list
+
+            scope_session.channels[""].channel_enabled = False
+            scope_session.channels[channel_names].configure_vertical(
+                vertical_range,
+                str_to_enum(VERTICAL_COUPLING_TO_ENUM, vertical_coupling),
+            )
+            scope_session.channels[channel_names].configure_chan_characteristics(
+                input_impedance, max_input_frequency=0.0
+            )
+            scope_session.configure_horizontal_timing(
+                min_sample_rate,
+                min_record_length,
+                ref_position=50.0,
+                num_records=1,
+                enforce_realtime=True,
             )
 
-        session_info = reservation.session_info[0]
-        channel_names = session_info.channel_list
-        pin_to_channel = {
-            mapping.pin_or_relay_name: mapping.channel for mapping in session_info.channel_mappings
-        }
-        if trigger_source in pin_to_channel:
-            trigger_source = pin_to_channel[trigger_source]
```

Comment:
Lines 143-147 are the only part of this example that can benefit from your new function.